src/core_plugins/kibana/public/home/components/tutorial/tutorial.js

```diff
@@ -0,0 +1,187 @@
+  ON_PREM_ELASTIC_CLOUD: 'onPremElasticCloud'
+};
+
+export class Tutorial extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      notFound: false,
+      paramValues: {},
+      tutorial: null
+    };
+
+    if (props.isCloudEnabled) {
+      this.state.visibleInstructions = INSTRUCTIONS_TYPE.ELASTIC_CLOUD;
+    } else {
+      this.state.visibleInstructions = INSTRUCTIONS_TYPE.ON_PREM;
+    }
+  }
+
+  async componentWillMount() {
+    const tutorial = await this.props.getTutorial(this.props.tutorialId);
+    if (tutorial) {
+      this.setState({
+        tutorial: tutorial
+      }, this.setParamDefaults);
+    } else {
+      this.setState({
+        notFound: true,
+      });
+    }
+  }
+
+  getInstructions = () => {
+    if (!this.state.tutorial) {
+      return { instructionSets: [] };
+    }
+
+    switch(this.state.visibleInstructions) {
+      case INSTRUCTIONS_TYPE.ELASTIC_CLOUD:
+        return this.state.tutorial.elasticCloud;
+      case INSTRUCTIONS_TYPE.ON_PREM:
+        return this.state.tutorial.onPrem;
+      case INSTRUCTIONS_TYPE.ON_PREM_ELASTIC_CLOUD:
+        return this.state.tutorial.onPremElasticCloud;
+      default:
+        throw new Error(`Unhandled instruction type ${this.state.visibleInstructions}`);
+    }
+  }
+
+  setParamDefaults = () => {
+    const instructions = this.getInstructions();
+    const paramValues = {};
+    if (instructions.params) {
+      instructions.params.forEach((param => {
+        paramValues[param.id] = param.defaultValue;
+      }));
+    }
+    this.setState({
+      paramValues: paramValues
+    });
+  }
+
+  setVisibleInstructions = (instructionsType) => {
+    this.setState({
+      visibleInstructions: instructionsType
+    }, this.setParamDefaults);
+  }
+
+  setParameter = (paramId, newValue) => {
+    this.setState(previousState => {
+      const paramValues = _.cloneDeep(previousState.paramValues);
+      paramValues[paramId] = newValue;
+      return { paramValues: paramValues };
+    });
+  }
+
+  onPrem = () => {
+    this.setVisibleInstructions(INSTRUCTIONS_TYPE.ON_PREM);
+  }
+
+  onPremElasticCloud = () => {
+    this.setVisibleInstructions(INSTRUCTIONS_TYPE.ON_PREM_ELASTIC_CLOUD);
+  }
+
+  renderInstructionSetsToggle = () => {
+    if (!this.props.isCloudEnabled) {
+      const radioButtons = [
+        { onClick: this.onPrem, label: 'On premise', dataTestSubj: 'onPremBtn' },
+        { onClick: this.onPremElasticCloud, label: 'Elastic cloud', dataTestSubj: 'onPremElasticCloudBtn' },
+      ];
+      return (
+        <RadioButtonGroup
+          buttons={radioButtons}
+          selectedBtnLabel={radioButtons[0].label}
+        />
+      );
+    }
+  }
+
+  renderInstructionSets = (instructions) => {
+    let offset = 1;
+    return instructions.instructionSets.map((instructionSet, index) => {
+      const currentOffset = offset;
+      offset += instructionSet.instructionVariants[0].instructions.length;
```

Comment:
Yes, instruction sets may have more than array of 1. APM is an example but others may was well as the specs get filled out

Reply:
Does APM have multiple sets of OS instructions too, per instruction set?  The offset logic doesn't really make sense in a generic sense.  E.g. if I'm on the second tab of the first set, and the first tab on the second set, the numbers won't necessarily line up;

<img width="657" alt="screen shot 2017-12-18 at 3 27 04 pm" src="https://user-images.githubusercontent.com/16563603/34126567-058c3018-e408-11e7-9633-9060f3b0ed11.png">

I don't really know how it makes sense if the user is looking at one variant for one set and another variant for the second set, because even if the numbers lined up, you wouldn't be following two different variants in two different sets.